_upgrading/sopel-8.md

```diff
@@ -0,0 +1,420 @@
+
+### Removed built-in plugins
+
+Sopel's built-in plugins are slowly being migrated out to their own standalone
+packages, which will help the project manage responsibility for maintenance and
+updates in the long term. In 8.0, we say farewell to:
+
+* `help`: now published as [`sopel-help`][sopel-help]
+  * Note: Sopel still requires the `sopel-help` package, so it will be installed
+    and available automatically.
+* `ip`: now published as [`sopel-iplookup`][sopel-iplookup]
+* `meetbot`: now published as [`sopel-meetbot`][sopel-meetbot]
+* `py`: now published as [`sopel-py`][sopel-py]
+* `reddit`: now published as [`sopel-reddit`][sopel-reddit]
+* `remind`: now published as [`sopel-remind`][sopel-remind]
+
+  [sopel-help]: https://pypi.org/project/sopel-help/
+  [sopel-iplookup]: https://pypi.org/project/sopel-iplookup/
+  [sopel-meetbot]: https://pypi.org/project/sopel-meetbot/
+  [sopel-py]: https://pypi.org/project/sopel-py/
+  [sopel-reddit]: https://pypi.org/project/sopel-reddit/
+  [sopel-remind]: https://pypi.org/project/sopel-remind/
+
+### Notable changes to built-in plugins
+
+#### `currency` plugin
+
+The `fiat_provider` setting now takes precedence over the `fixer_io_key`.
+Previously, setting a `fixer_io_key` would use the `fixer.io` fiat exchange rate
+provider regardless of the `fiat_provider` setting.
+
+The old `fiat_provider` default of `exchangerate.host` is no longer a valid
+choice; it has been replaced by `open.er-api.com`.
+
+#### `reload` plugin
+
+The `.update` command has been removed from the `reload` plugin. Its functioning
+relied on running Sopel in a way that isn't officially supported, so we chose to
+remove it.
+
+#### `wikipedia` plugin
+
+The old command names (`.w`, `.wik`, and `.wiki`) are no longer used; they were
+freed up for use by bespoke plugins. `wikipedia` functionality is now invoked by
+the `.wikipedia` command, or `.wp` for short.
+
+The deprecated `lang_per_channel` config-file setting has been removed.
+
+
+## Sopel 8 API changes
+
+This is a good time to remind you that your plugins can specify both minimum and
+maximum Sopel versions in their requirements. If all plugins a bot owner wishes
+to install contain this metadata, they can ask `pip` to install Sopel and all of
+its plugins in a single command, and `pip`'s dependency solver will figure out
+which version of Sopel satisfies all of the plugins' needs.
+
+We do everything we can to keep breaking API changes to *only* major versions,
+so a version range like `sopel>=7.1,<9` is typically safe—at least the earliest
+Sopel version your plugin supports, up to (but excluding) the next unreleased
+major version.
+
+In the vast majority of cases, removed API features will first go through a
+deprecation period, during which Sopel will log warnings whenever the deprecated
+functionality is used by a plugin. Rarely, we might need to remove an API
+feature without going through this process—but that's a last-resort option.
```

Comment:
Overall, this section feels more like general guidance on plugin authorship than it does a guide to migrating between <8 and 8.0.

Reply:
Before I get to the other comments in this section (the lede of "Sopel 8 API changes"), it might indeed make more sense to just drop it and go directly from the heading to the first subsection:

```md
## Sopel 8 API changes

### Command names are now literal
```

Reply:
I think I agree yes. Do we need a way to talk about plugin management in some way or another? Yes. Is it the right place to do so? Maybe not. A lot of good ideas in here in any case.

Reply:
I removed this section from the newest upgrade guide revision, because I agree this isn't the place to talk about plugin versioning & dependency management. sopel-irc/sopel#2604 will carry the torch from here.